File: WebApp/indoor-localization/src/entities/AssetZoneHistoryLog.ts
export class AssetZoneHistoryLog {
  id: number;
  assetId: number;
  zoneId: number;
  enterDateTime: Date;
  exitDateTime: Date;
  retentionTime: number;

  constructor(
    id: number,
    assetId: number,
    zoneId: number,
    enterDateTime: Date,
    exitDateTime: Date,
    retentionTime: number
  ) {
    this.id = id;
    this.assetId = assetId;
    this.zoneId = zoneId;
    this.enterDateTime = enterDateTime;
    this.exitDateTime = exitDateTime;
    this.retentionTime = retentionTime;
  }
}

File: Android/indoor_localization/lib/domain/entities/asset_zone_history_log.dart
import 'package:indoor_localization/domain/entities/zone.dart';
import 'Asset.dart';

class AssetZoneHistoryLog {
  final int id;
  final Asset asset;
  final Zone zone;
  final DateTime enterDateTime;
  final DateTime exitDateTime;
  final int retentionTime;

  AssetZoneHistoryLog({
    required this.id,
    required this.asset,
    required this.zone,
    required this.enterDateTime,
    required this.exitDateTime,
    required this.retentionTime,
  });
}
